### DISCUSSION TITLE
How to use Virtual Components in Typescript?

### DISCUSSION THREAD

**Discussion by steveatkoan** [vuetifyjs/vuetify#17485] [Q&A]
Hi

I'd like to set up virtual components https://vuetifyjs.com/en/features/global-configuration/#using-with-virtual-components but WebStorm reports the component as an unknown tag in a Single File Component. How do I tell typescript / WebStorm the component exists?

```
const vuetify = createVuetify({
  aliases: {
    VBtnPrimary: VBtn,
  },
  defaults: {
    VBtn: {
      class: ['text-none']
    },
    VBtnPrimary: {
      color: 'primary',
      variant: 'flat',
    }
  }
});
```

```
<template>
  <v-btn-primary text="Save" />
</template>
```

**Comment by mj2068** [comment]
yeah, i use vscode. i wonder is it even possible to provide editor support for virtual components?

**Comment by KaelWD** [comment]
```ts
declare module '@vue/runtime-core' {
  export interface GlobalComponents {
    VBtnPrimary: typeof import('vuetify/components/VBtn')['VBtn']
  }
}
```

**Reply by mj2068** [reply]
@KaelWD awesome, thank you Kael.

**Reply by pastukhvladyslav** [reply]
It didn't work for me. This is what worked for me (from [another issue](https://github.com/vuetifyjs/vuetify/issues/18552#issuecomment-1806399966)):
```
declare module '@vue/runtime-dom' {
  export interface GlobalComponents {
    VBtnPrimary: typeof import('vuetify/components')['VBtn']
  }
}

export {}
```
